py/objtuple.c

```diff
@@ -102,7 +106,7 @@ STATIC mp_obj_t mp_obj_tuple_make_new(const mp_obj_type_t *type_in, size_t n_arg
 // Don't pass MP_BINARY_OP_NOT_EQUAL here
 STATIC mp_obj_t tuple_cmp_helper(mp_uint_t op, mp_obj_t self_in, mp_obj_t another_in) {
     // type check is done on getiter method to allow tuple, namedtuple, attrtuple
-    mp_check_self(mp_obj_get_type(self_in)->getiter == mp_obj_tuple_getiter);
+    assert(_mp_obj_is_tuple_compatible(self_in));
```

Comment:
This should actually stay as a call to `mp_check_self(...)` because the incoming object can be supplied by a user from a script.  It may already be assumed to be a concrete object (ie a pointer) but that doesn't mean it can just be an assert here.